firmware/util/loggingcentral.cpp

```diff
@@ -63,11 +63,10 @@ void LogBuffer<TBufferSize>::writeInternal(const char* buffer) {
 
 	// If we can't fit the whole thing, write as much as we can
 	len = minI(available, len);
+	// Ensure the output buffer is always null terminated (in case we did a partial write)
+	*(m_writePtr + len) = '\0';
```

Comment:
Just understand that this can cause write out on array bounds. Or not? Am I paranoid?
I `len` was clamped by `available` one line above. Then `m_writePtr + len` is outside buffer...

Edited: I need to carefully read comments. Everything is ok